### PR TITLE
Add selected commit to repo persistence

### PR DESCRIPTION
**Pull request by mariusraht2**
Related to #3040

Add checked out commit to repo persistence. Changed also related getter and setter.

### CODE REVIEW

File: src/zcl_abapgit_repo_online.clas.abap [hunk 1]
```diff
@@ -337,7 +337,7 @@ CLASS ZCL_ABAPGIT_REPO_ONLINE IMPLEMENTATION.
 
   METHOD select_commit.
     reset_remote( ).
-    mv_current_commit = iv_sha1.
+    set( iv_selected_commit = iv_selected_commit ).
```
Comment (larshp) [inline src/zcl_abapgit_repo_online.clas.abap:340]: this should also reset the branch name, so the code can know its now a commie SHA1 that the user has chosen?

**Comment by mariusraht2** [comment]
Do we have to adjust `switch_origin` in `zcl_abapgit_repo_online` too?

File: src/zcl_abapgit_repo_online.clas.abap (hunk 1 above)
Comment (mariusraht2) [reply src/zcl_abapgit_repo_online.clas.abap:340]: ...and vice versa.

Comment (larshp) [reply src/zcl_abapgit_repo_online.clas.abap:340]: yes

**Comment by larshp** [comment]
switch_origin: probably, it should reset the selected_commit

**Comment by mariusraht2** [comment]
Hmm...just seeing that it calls `select_branch` if origin has to be switched. That seems ok.

**Comment by mariusraht2** [comment]
Shouldn't empty branch_name though:
image

**Comment by mariusraht2** [comment]
I need it to determine the commits to be selectable by the user

**Pull request merged**